### ISSUE TITLE
Style/NumericPredicate broke `$CHILD_STATUS == 0`

### ISSUE THREAD

**Issue by ShockwaveNN** [bbatsov/rubocop#3892]
## Steps to reproduce the problem
Create temp.rb file with
```
require "English"
pid = spawn('exit 0')
Process.wait pid
puts $CHILD_STATUS == 0
```
Run `ruby temp.rb #true `
Run `rubocop temp.rb`

It show warning
```
temp.rb:3:6: C: Style/NumericPredicate: Use $CHILD_STATUS.zero? instead of $CHILD_STATUS == 0
```
If you fix it to
```
require "English"
pid = spawn('exit 0')
Process.wait pid
puts $CHILD_STATUS.zero?
```
It broke
```
ruby temp.rb
temp.rb:4:in `<main>': undefined method `zero?' for #<Process::Status: pid 13846 exit 0> (NoMethodError)
```
## RuboCop version

```
$ rubocop -V
0.46.0 (using Parser 2.3.3.1, running on ruby 2.1.10 x86_64-linux)
```

**Comment by Drenmi** [comment]
Yes. This cop has been known to cause some false positives. The reason being we can't know what type the receiver will be at runtime (since we're doing static code analysis. 😀) We make the assumption that if a receiver is compared with `0`, that receiver is a duck type of `Numeric`.

You can disable this for the line affected only using:

```
if not_a_numeric == 0 # rubocop:disable Style/NumericPredicate
```

or, if this is causing enough false positives that it's not worth it, you can disable it entirely in your `.rubocop.yml` file:

```
Style/NumericPredicate:
  Enabled: false
```

😊

**Comment by mikegee** [comment]
For the particular example you posted, [`Process::Status#success?`](https://ruby-doc.org/core-2.2.2/Process/Status.html#method-i-success-3F) is an improvement over either sort of numeric comparison.

`Process::Status` allows numeric comparison to be compatible with code written a very long time ago. `$CHILD_STATUS` has been an instance of `Process::Status` since at least Ruby 1.8.6.

**Comment by bbatsov** [comment]
> Process::Status allows numeric comparison to be compatible with code written a very long time ago. $CHILD_STATUS has been an instance of Process::Status since at least Ruby 1.8.6.
>  👍 2

We should definitely add a cop for this - it'd be trivial to implement.
